ocrd/ocrd/resource_manager.py

```diff
@@ -137,7 +137,7 @@ def default_resource_dir(self):
     def location_to_resource_dir(self, location):
         return '/usr/local/share/ocrd-resources' if location == 'system' else \
                 join(XDG_DATA_HOME, 'ocrd-resources') if location == 'data' else \
-                getcwd()
+                join(getcwd(), 'ocrd-resources')
```

Comment:
That is totally unexpected IMHO. CWD-relative path names should not get any prefix directory, neither `ocrd-resources` nor the processor name.

Reply:
> With the 2.22.3 behavior, it consistently stores and looks in `$PWD/ocrd-resources>/<executable>/<resname>`.
> 
> And I honestly thought that's how it should be but indeed the specs say `$PWD/<resname>`.

And so I (strongly) feel they should :slightly_smiling_face: 

> We can fix this by `pushd_popd(self.old_pwd)` before we check whether `<resname>` can be resolved as-is. That would still allow you to drop resources flat into `$PWD`.

Whatever it takes!

> One reason I find `$PWD/ocrd-resources` useful is for caching in CI, because you don't need to list all the individual files like that.

I don't quite understand that TBH...

Reply:
> > With the 2.22.3 behavior, it consistently stores and looks in `$PWD/ocrd-resources>/<executable>/<resname>`.
> > And I honestly thought that's how it should be but indeed the specs say `$PWD/<resname>`.
> 
> And so I (strongly) feel they should 🙂
> 
> > We can fix this by `pushd_popd(self.old_pwd)` before we check whether `<resname>` can be resolved as-is. That would still allow you to drop resources flat into `$PWD`.
> 
> Whatever it takes!

OK, but that only solves half of the problem, the other question is whether `ocrd resmgr --location cwd` should download a resource `<res>` to `$PWD/<res>` or `$PWD/ocrd-resources/<res>`. IIUC you also want the download to the former, without `ocrd-resources`, correct?

> 
> > One reason I find `$PWD/ocrd-resources` useful is for caching in CI, because you don't need to list all the individual files like that.
> 
> I don't quite understand that TBH...

>> One reason I find $PWD/ocrd-resources useful is for caching in CI, because you don't need to list all the individual files like that.

> I don't quite understand that TBH...

We want to cache models across CI runs so we don't need to download gigabytes of model data every time. If all the models are in one directory `$PWD/ocrd-resources`, you can just cache that whole directory instead of listing explicitly every file in `$PWD` thart should be cached. It's a minor thing.

Reply:
> OK, but that only solves half of the problem, the other question is whether `ocrd resmgr --location cwd` should download a resource `<res>` to `$PWD/<res>` or `$PWD/ocrd-resources/<res>`. IIUC you also want the download to the former, without `ocrd-resources`, correct?

Yes.

> We want to cache models across CI runs so we don't need to download gigabytes of model data every time. If all the models are in one directory `$PWD/ocrd-resources`, you can just cache that whole directory instead of listing explicitly every file in `$PWD` thart should be cached. It's a minor thing.

Oh, I got it. But you could still have the entire CWD cached, if you placed the workspace elsewhere.